packages/react/alert-dialog/src/AlertDialog.tsx

```diff
@@ -47,112 +34,44 @@ const [
   useAlertDialogContentContext,
 ] = createContext<AlertDialogContentContextValue>(CONTENT_NAME);
 
-type AlertDialogContentOwnProps = Omit<
-  Polymorphic.OwnProps<typeof DialogPrimitive.Content>,
-  'refToFocusOnOpen' | 'id'
->;
+type AlertDialogContentOwnProps = Omit<Polymorphic.OwnProps<typeof DialogPrimitive.Content>, 'id'>;
 
 type AlertDialogContentPrimitive = Polymorphic.ForwardRefComponent<
   Polymorphic.IntrinsicElement<typeof DialogPrimitive.Content>,
   AlertDialogContentOwnProps
 >;
 
 const AlertDialogContent = React.forwardRef((props, forwardedRef) => {
-  const {
-    'aria-label': ariaLabel,
-    'aria-labelledby': ariaLabelledBy,
-    'aria-describedby': ariaDescribedBy,
-    children,
-    ...dialogContentProps
-  } = props;
-  const { descriptionId, titleId } = useAlertDialogContext(CONTENT_NAME);
   const cancelRef = React.useRef<React.ElementRef<typeof AlertDialogCancel> | null>(null);
 
   return (
-    <DialogPrimitive.Content
-      role="alertdialog"
-      aria-describedby={ariaDescribedBy || descriptionId}
-      // If `aria-label` is set, ensure `aria-labelledby` is undefined as to avoid confusion.
-      // Otherwise fallback to an explicit `aria-labelledby` or the ID used in the
-      // `AlertDialogTitle`
-      aria-labelledby={ariaLabel ? undefined : ariaLabelledBy || titleId}
-      aria-label={ariaLabel || undefined}
-      {...dialogContentProps}
-      ref={forwardedRef}
-      onOpenAutoFocus={composeEventHandlers(dialogContentProps.onOpenAutoFocus, (event) => {
-        event.preventDefault();
-        cancelRef.current?.focus({ preventScroll: true });
-      })}
+    <DialogPrimitive.AccessibilityDevWarningsContext.Provider
+      value={React.useMemo(
+        () => ({
+          descriptionRequired: true,
+          partNames: { content: CONTENT_NAME, title: TITLE_NAME, description: DESCRIPTION_NAME },
+          docsUrl: 'https://radix-ui.com/primitives/docs/components/alert-dialog',
+        }),
+        []
+      )}
```

Comment:
I've used a React context defined in `Dialog` with a default value, which we override here in `AlertDialog` to provide the necessary details for this specific primitive requirements.

Let me know if you can find another way, I started originally going via props, but it bothered me to have "private" props exposed publicly just because they were needed to be passed from `AlertDialog`'s internals into `Dialog`.

Interestingly, this could maybe be a way to solve the longstanding #210.

Reply:
One thing to note here is that this particular bit of code here won't be dead-eliminated in prod, let me know if you'd like me to find a way to remodel this conditionally to only include it in dev or if you think it's unnecessary complexity.